recipes/cpython/all/conanfile.py

```diff
@@ -53,286 +56,294 @@ class CPythonConan(ConanFile):
-            conf_args.append("--without-gcc")
+        if Version(self.version) >= "3.10":
+            tc.configure_args.append("--disable-test-modules")
+        if self.options.get_safe("with_sqlite3"):
+            tc.configure_args.append("--enable-loadable-sqlite-extensions={}".format(
+                yes_no(not self.dependencies["sqlite3"].options.omit_load_extension)
+            ))
+        if self.settings.compiler == "intel-cc":
+            tc.configure_args.append("--with-icc")
+        if os.environ.get("CC") or self.settings.compiler != "gcc":
+            tc.configure_args.append("--without-gcc")
         if self.options.with_tkinter:
             tcltk_includes = []
             tcltk_libs = []
             # FIXME: collect using some conan util (https://github.com/conan-io/conan/issues/7656)
             for dep in ("tcl", "tk", "zlib"):
-                tcltk_includes += ["-I{}".format(d) for d in self.deps_cpp_info[dep].include_paths]
-                tcltk_libs += ["-l{}".format(lib) for lib in self.deps_cpp_info[dep].libs]
-            if self.settings.os == "Linux" and not self.options["tk"].shared:
+                cpp_info = self.dependencies[dep].cpp_info.aggregated_components()
+                tcltk_includes += [f"-I{d}" for d in cpp_info.includedirs]
+                tcltk_libs += [f"-L{lib}" for lib in cpp_info.libdirs]
+                tcltk_libs += [f"-l{lib}" for lib in cpp_info.libs]
+            if self.settings.os in ["Linux", "FreeBSD"] and not self.dependencies["tk"].options.shared:
                 # FIXME: use info from xorg.components (x11, xscrnsaver)
-                tcltk_libs.extend(["-l{}".format(lib) for lib in ("X11", "Xss")])
-            conf_args.extend([
+                tcltk_libs.extend([f"-l{lib}" for lib in ("X11", "Xss")])
+            tc.configure_args += [
                 "--with-tcltk-includes={}".format(" ".join(tcltk_includes)),
                 "--with-tcltk-libs={}".format(" ".join(tcltk_libs)),
-            ])
-        if self.settings.os in ("Linux", "FreeBSD"):
-            # Building _testembed fails due to missing pthread/rt symbols
-            self._autotools.link_flags.append("-lpthread")
-
-        build = None
-        if tools.cross_building(self) and not tools.cross_building(self, skip_x64_x86=True):
-            # Building from x86_64 to x86 is not a "real" cross build, so set build == host
-            build = tools.get_gnu_triplet(str(self.settings.os), str(self.settings.arch), str(self.settings.compiler))
-        self._autotools.configure(args=conf_args, configure_dir=self._source_subfolder, build=build)
-        return self._autotools
+            ]
+        if not is_apple_os(self):
+            tc.extra_ldflags.append('-Wl,--as-needed')
+
+        tc.generate()
+
+        deps = AutotoolsDeps(self)
+        deps.generate()
+
+    def generate(self):
+        VirtualRunEnv(self).generate(scope="build")
```

Comment:
Which requirement do you need here, that requires VirtualRunEnv?

Reply:
The build runs Python once it's done to do some sanity checks, so we need access to the run environment in the build.